core/primitives/src/congestion_info.rs

```diff
@@ -302,14 +233,68 @@ impl CongestionInfo {
         Ok(())
     }
 
-    /// TODO(congestion_info) - Fix all caller-sites and remove this function
+    /// Congestion level ignoring the chain context (missed chunks count).
+    pub fn localized_congestion_level(&self, config: &CongestionControlConfig) -> f64 {
+        let incoming_congestion = self.incoming_congestion(config);
+        let outgoing_congestion = self.outgoing_congestion(config);
+        let memory_congestion = self.memory_congestion(config);
+        incoming_congestion.max(outgoing_congestion).max(memory_congestion)
+    }
+
+    pub fn incoming_congestion(&self, config: &CongestionControlConfig) -> f64 {
+        clamped_f64_fraction(self.delayed_receipts_gas(), config.max_congestion_incoming_gas)
+    }
+
+    pub fn outgoing_congestion(&self, config: &CongestionControlConfig) -> f64 {
+        clamped_f64_fraction(self.buffered_receipts_gas(), config.max_congestion_outgoing_gas)
+    }
+
+    pub fn memory_congestion(&self, config: &CongestionControlConfig) -> f64 {
+        clamped_f64_fraction(self.receipt_bytes() as u128, config.max_congestion_memory_consumption)
+    }
+
+    /// Computes and sets the `allowed_shard` field.
+    ///
+    /// If in a fully congested state, decide which shard of `other_shards` is
+    /// allowed to forward to `own_shard` this round. In this case, we stop all
+    /// of `other_shards` from sending anything to `own_shard`. But to guarantee
+    /// progress, we allow one shard of `other_shards` to send `RED_GAS` in the
```

Comment:
nit: RED_GAS is so legacy now... ;) 
Totally my bad for missing that when I renamed it :innocent:

Reply:
good catch :)